wdl-lint/tests/lints/matching-param-meta/source.errors

```diff
@@ -1,32 +1,91 @@
+   ┌─ tests/lints/matching-param-meta/source.wdl:28:16
    │
-25 │         String does_not_exist
+28 │         String does_not_exist
    │                ^^^^^^^^^^^^^^ this input does not have an entry in the parameter metadata section
    │
    = fix: add a `does_not_exist` key to the `parameter_meta` section with a detailed description of the input.
 
 note[MatchingParameterMeta]: workflow `w` has an extraneous parameter metadata key named `extra`
-   ┌─ tests/lints/matching-param-meta/source.wdl:47:9
+   ┌─ tests/lints/matching-param-meta/source.wdl:48:9
    │
-47 │         extra: "this should not be here"
+48 │         extra: "this should not be here"
    │         ^^^^^ this key does not correspond to any input declaration
    │
    = fix: remove the extraneous key from the `parameter_meta` section
 
 warning[MatchingParameterMeta]: workflow `w` is missing a parameter metadata key for input `does_not_exist`
-   ┌─ tests/lints/matching-param-meta/source.wdl:52:16
+   ┌─ tests/lints/matching-param-meta/source.wdl:53:16
    │
-52 │         String does_not_exist
+53 │         String does_not_exist
    │                ^^^^^^^^^^^^^^ this input does not have an entry in the parameter metadata section
    │
    = fix: add a `does_not_exist` key to the `parameter_meta` section with a detailed description of the input.
 
+warning[MatchingParameterMeta]: parameter metadata in task `new_test` is out of order
+   ┌─ tests/lints/matching-param-meta/source.wdl:64:5
+   │
+64 │     parameter_meta {
+   │     ^^^^^^^^^^^^^^ parameter metadata must be in the same order as inputs
+   │
+   = fix: order the parameter metadata as:
+     first
+     second
+
+warning[MatchingParameterMeta]: parameter metadata in struct `Bar` is out of order
+   ┌─ tests/lints/matching-param-meta/source.wdl:83:5
+   │
+83 │     parameter_meta {
+   │     ^^^^^^^^^^^^^^ parameter metadata must be in the same order as inputs
+   │
+   = fix: order the parameter metadata as:
+     param_a
+     param_b
+
+note[InputSorting]: input not sorted
+    ┌─ tests/lints/matching-param-meta/source.wdl:104:5
+    │
+104 │     input {
+    │     ^^^^^ input section must be sorted
+    │
+    = fix: sort input statements as: 
+      File b
+      File t
+      Directory w
+      Array[String]+ p
+      Array[String]+ q
+
+warning[MatchingParameterMeta]: parameter metadata in task `input_sorting_test_2` is out of order
+    ┌─ tests/lints/matching-param-meta/source.wdl:122:5
+    │
+122 │     parameter_meta {
+    │     ^^^^^^^^^^^^^^ parameter metadata must be in the same order as inputs
+    │
+    = fix: order the parameter metadata as:
+      w
+      p
+      t
+      q
+      b
+
+note[InputSorting]: input not sorted
+    ┌─ tests/lints/matching-param-meta/source.wdl:130:5
+    │
+130 │     input {
+    │     ^^^^^ input section must be sorted
+    │
+    = fix: sort input statements as: 
+      File t
+      File b
+      Directory w
+      Array[String]+ p
+      Array[String]+ q
```

Comment:
This is potentially confusing. I'm not sure what the best behavior is. We might want to silence the "out of order" diagnostic from `MatchingParameterMeta` if `InputSorting` is firing, but I'm not sure the right way to architect that. We would still want the "extra" and "missing" diagnostics from `MatchingParameterMeta` regardless of the parameter meta sorting.

Tagging @peterhuene and @adthrasher to offer their two cents on the best behavior.